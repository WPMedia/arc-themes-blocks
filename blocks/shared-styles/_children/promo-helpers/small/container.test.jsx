import React from 'react';
import { mount } from 'enzyme';
import SmallPromoContainer from './container';

describe('SmallPromoContainer', () => {
  const mockHeadline = (<h1>mockHeadline</h1>);
  const mockImage = (<img href="mocklink.jpg" alt="mock image" />);

  it('renders with no image', () => {
    const result = mount(<SmallPromoContainer headline={mockHeadline} imagePosition="above" />);
    expect(result).toMatchSnapshot();
  });

  it('default to headline first dom structure by default', () => {
    const result = mount(<SmallPromoContainer headline={mockHeadline} image={mockImage} />);
    expect(result).toMatchSnapshot();
  });

  it('render headline first if image position is below', () => {
    const result = mount(<SmallPromoContainer headline={mockHeadline} image={mockImage} imagePosition="below" />);
    expect(result).toMatchSnapshot();
  });

  it('render image first if image position is on left', () => {
    const result = mount(<SmallPromoContainer headline={mockHeadline} image={mockImage} imagePosition="left" />);
    expect(result).toMatchSnapshot();
  });

  it('render image first if image position is above', () => {
    const result = mount(<SmallPromoContainer headline={mockHeadline} image={mockImage} imagePosition="above" />);
    expect(result).toMatchSnapshot();
  });
});
