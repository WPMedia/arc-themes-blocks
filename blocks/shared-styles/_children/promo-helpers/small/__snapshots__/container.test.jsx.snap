// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallPromoContainer default to headline first dom structure by default 1`] = `
<SmallPromoContainer
  headline={
    <h1>
      mockHeadline
    </h1>
  }
  image={
    <img
      alt="mock image"
      href="mocklink.jpg"
    />
  }
>
  <article
    className="container-fluid small-promo"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-xl-8"
      >
        <h1>
          mockHeadline
        </h1>
      </div>
      <div
        className="col-sm-xl-4"
      >
        <img
          alt="mock image"
          href="mocklink.jpg"
        />
      </div>
    </div>
  </article>
  <hr />
</SmallPromoContainer>
`;

exports[`SmallPromoContainer render headline first if image position is below 1`] = `
<SmallPromoContainer
  headline={
    <h1>
      mockHeadline
    </h1>
  }
  image={
    <img
      alt="mock image"
      href="mocklink.jpg"
    />
  }
  imagePosition="below"
>
  <article
    className="container-fluid small-promo"
  >
    <div
      className="col-sm-xl-8"
    >
      <h1>
        mockHeadline
      </h1>
    </div>
    <div
      className="col-sm-xl-4"
    >
      <img
        alt="mock image"
        href="mocklink.jpg"
      />
    </div>
  </article>
  <hr />
</SmallPromoContainer>
`;

exports[`SmallPromoContainer render image first if image position is above 1`] = `
<SmallPromoContainer
  headline={
    <h1>
      mockHeadline
    </h1>
  }
  image={
    <img
      alt="mock image"
      href="mocklink.jpg"
    />
  }
  imagePosition="above"
>
  <article
    className="container-fluid small-promo"
  >
    <div
      className="col-sm-xl-4"
    >
      <img
        alt="mock image"
        href="mocklink.jpg"
      />
    </div>
    <div
      className="col-sm-xl-8"
    >
      <h1>
        mockHeadline
      </h1>
    </div>
  </article>
  <hr />
</SmallPromoContainer>
`;

exports[`SmallPromoContainer render image first if image position is on left 1`] = `
<SmallPromoContainer
  headline={
    <h1>
      mockHeadline
    </h1>
  }
  image={
    <img
      alt="mock image"
      href="mocklink.jpg"
    />
  }
  imagePosition="left"
>
  <article
    className="container-fluid small-promo"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-xl-4"
      >
        <img
          alt="mock image"
          href="mocklink.jpg"
        />
      </div>
      <div
        className="col-sm-xl-8"
      >
        <h1>
          mockHeadline
        </h1>
      </div>
    </div>
  </article>
  <hr />
</SmallPromoContainer>
`;

exports[`SmallPromoContainer renders with no image 1`] = `
<SmallPromoContainer
  headline={
    <h1>
      mockHeadline
    </h1>
  }
  imagePosition="above"
>
  <article
    className="container-fluid small-promo"
  >
    <div
      className="col-sm-xl-4"
    />
    <div
      className="col-sm-xl-12"
    >
      <h1>
        mockHeadline
      </h1>
    </div>
  </article>
  <hr />
</SmallPromoContainer>
`;
